tests/test_clang_delta.py

```diff
@@ -32,3 +35,289 @@ async def equals_source(x):
+    from unittest.mock import patch
+
+    # Mock both which and glob to return nothing
+    with patch("shrinkray.passes.clangdelta.which", return_value=None):
+        with patch("shrinkray.passes.clangdelta.glob", return_value=[]):
+            result = find_clang_delta()
+            assert result == ""
+
+
+async def test_query_instances_raises_clang_delta_error():
+    """Test query_instances raises ClangDeltaError for non-assertion errors.
+
+    This tests line 133: raise ClangDeltaError(msg) when CalledProcessError
+    doesn't contain 'Assertion failed'.
+    """
+    import subprocess
+    from unittest.mock import patch
+
+    cd = ClangDelta(find_clang_delta())
+
+    # Mock trio.run_process to raise CalledProcessError without "Assertion failed"
+    error = subprocess.CalledProcessError(1, "clang_delta")
+    error.stdout = b"Some other error"
+    error.stderr = b"More error info"
+
+    with patch("trio.run_process", side_effect=error):
+        with pytest.raises(ClangDeltaError):
+            await cd.query_instances("rename-var", b"int main() {}")
+
+
+async def test_apply_transformation_no_modification():
+    """Test apply_transformation returns original when 'No modification' error.
+
+    This tests line 167: return data when the error message indicates no modification.
+    """
+    import subprocess
+    from unittest.mock import patch
+
+    cd = ClangDelta(find_clang_delta())
+    source = b"int main() { return 0; }"
+
+    # Mock trio.run_process to raise CalledProcessError with "No modification" message
+    error = subprocess.CalledProcessError(1, "clang_delta")
+    error.stdout = b"Error: No modification to the transformed program!"
+    error.stderr = b""
+
+    with patch("trio.run_process", side_effect=error):
+        result = await cd.apply_transformation("rename-var", 1, source)
+        assert result == source
+
+
+async def test_pump_handles_query_instances_error():
+    """Test clang_delta_pump handles ClangDeltaError from query_instances.
+
+    This tests lines 190-191: ClangDeltaError during query_instances returns target.
+    """
+    from unittest.mock import patch
+
+    cd = ClangDelta(find_clang_delta())
+    pump = clang_delta_pump(cd, "rename-var")
+
+    source = b"int main() { return 0; }"
+
+    async def is_interesting(x):
+        return True
+
+    problem = BasicReductionProblem(
+        source, is_interesting, work=WorkContext(parallelism=1)
+    )
+
+    # Mock query_instances to raise ClangDeltaError
+    with patch.object(
+        cd, "query_instances", side_effect=ClangDeltaError(b"Query failed")
+    ):
+        result = await pump(problem)
+        assert result == source
+
+
+async def test_pump_makes_progress_when_transformation_is_interesting():
+    """Test clang_delta_pump makes progress when transformations produce interesting results.
+
+    This tests lines 215-217: the code path after find_first_value succeeds.
```

Comment:
Lines 215-217 are not a durable reference, do you mean test_find_clang_delta_when_not_in_path() ?